core/services/synchronization/explorer_client.go

```diff
@@ -130,12 +123,13 @@ func (ec *explorerClient) Start() error {
 		return nil
 	}
 
-	ctx, cancel := context.WithCancel(context.Background())
-	ec.cancel = cancel
+	ec.done = make(chan struct{})
+
 	wg := &sync.WaitGroup{}
 	wg.Add(1)
-	go ec.connectAndWritePump(ctx, wg)
```

Comment:
I'm not following why we even need this goroutine here? If we're going to wg.Wait() right after starting it we can just make a blocking call to connectAndWritePump or even inline that code here

Reply:
Yup, I get rid of this here: https://github.com/smartcontractkit/chainlink/pull/4141